### PR TITLE
[ticket/13455] Change request_var() calls with $request->variable()

### PR DESCRIPTION
**Pull request by MGaetan89**
https://tracker.phpbb.com/browse/PHPBB3-13455

PHPBB3-13455

### CODE REVIEW

**Comment by MGaetan89** [comment]
!set WIP

**Comment by nickvergessen** [comment]
Please use 1 PR per function, this PR is already big enough

**Comment by MGaetan89** [comment]
I was thinking the same thing. I'm just keeping track of deprecated functions here.

File: phpBB/includes/functions_compatibility.php [hunk 1]
```diff
@@ -195,3 +195,69 @@ function get_user_rank($user_rank, $user_posts, &$rank_title, &$rank_img, &$rank
 	$rank_img = $rank_data['img'];
 	$rank_img_src = $rank_data['img_src'];
 }
+
+/**
+ * Wrapper function of \phpbb\request\request::variable which exists for backwards compatability.
+ * See {@link \phpbb\request\request_interface::variable \phpbb\request\request_interface::variable} for
+ * documentation of this function's use.
+ *
+ * @deprecated 3.2.0-a1 (To be removed: 3.4.0)
```
Comment (MGaetan89) [inline phpBB/includes/functions_compatibility.php]: Can someone check if this is correct?

I moved this function to `functions_compatibility`, as it is no longer used in phpBB itself.

Comment (nickvergessen) [inline phpBB/includes/functions_compatibility.php]: It's already deprecated in 3.1, so remove in 3.3

File: phpBB/download/file.php [hunk 2]
```diff
@@ -80,7 +80,7 @@
 
 	unset($dbpasswd);
 
-	request_var('', 0, false, false, $request);
+	$request->variable('', 0, false, false, $request);
```
Comment (Nicofuma) [inline phpBB/download/file.php]: to be removed

File: phpBB/phpbb/session.php [hunk 3]
```diff
@@ -253,23 +253,23 @@ function session_begin($update_session_page = true)
 
 		if ($request->is_set($config['cookie_name'] . '_sid', \phpbb\request\request_interface::COOKIE) || $request->is_set($config['cookie_name'] . '_u', \phpbb\request\request_interface::COOKIE))
 		{
-			$this->cookie_data['u'] = request_var($config['cookie_name'] . '_u', 0, false, true);
-			$this->cookie_data['k'] = request_var($config['cookie_name'] . '_k', '', false, true);
-			$this->session_id 		= request_var($config['cookie_name'] . '_sid', '', false, true);
+			$this->cookie_data['u'] = $request->variable($config['cookie_name'] . '_u', 0, false, true);
```
Comment (Nicofuma) [inline phpBB/phpbb/session.php]: $request->variable($config['cookie_name'] . '_u', 0, false, \phpbb\request\request_interface::COOKIE)

File: phpBB/phpbb/session.php [hunk 4]
```diff
@@ -253,23 +253,23 @@ function session_begin($update_session_page = true)
...
+			$this->cookie_data['k'] = $request->variable($config['cookie_name'] . '_k', '', false, true);
```
Comment (Nicofuma) [inline phpBB/phpbb/session.php]: $request->variable($config['cookie_name'] . '_k', 0, false, \phpbb\request\request_interface::COOKIE)

File: phpBB/phpbb/session.php [hunk 5]
```diff
@@ -253,23 +253,23 @@ function session_begin($update_session_page = true)
...
+			$this->session_id 		= $request->variable($config['cookie_name'] . '_sid', '', false, true);
```
Comment (Nicofuma) [inline phpBB/phpbb/session.php]: $request->variable($config['cookie_name'] . '_sid', 0, false, \phpbb\request\request_interface::COOKIE)

File: phpBB/includes/functions.php [hunk 6]
```diff
@@ -1707,7 +1641,7 @@ function get_unread_topics($user_id = false, $sql_extra = '', $sql_sort = '', $s
 
 		if (empty($tracking_topics))
 		{
-			$tracking_topics = request_var($config['cookie_name'] . '_track', '', false, true);
+			$tracking_topics = $request->variable($config['cookie_name'] . '_track', '', false, true);
```
Comment (Nicofuma) [inline phpBB/includes/functions.php]: $request->variable($config['cookie_name'] . '_track', '', false, \phpbb\request\request_interface::COOKIE)

**Comment by MGaetan89** [comment]
Thanks @Nicofuma, I updated the code.

**Comment by MGaetan89** [comment]
!unset WIP

**Comment by MGaetan89** [comment]
Bump

**Comment by MGaetan89** [comment]
Rebased on `develop`.

**Comment by bantu** [comment]
This should be good to, but also conflicted in the meantime. Please rebase.

As a follow up PR: All utf8_normalize_nfc() calls around $request->variable() can be removed as this function is already called by $request.

**Comment by MGaetan89** [comment]
Can't I remove `utf8_normalize_nfc()` in this PR?
IMO, it is part of the update process.

**Comment by bantu** [comment]
@MGaetan89 Yeah, I agree. Sounds good.

**Comment by nickvergessen** [comment]
missing the ticket prefix on the last commit

**Comment by MGaetan89** [comment]
Fixed.

**Pull request merged**